config/pkcs11/config.go

```diff
@@ -0,0 +1,116 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package pkcs11config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/containers/ocicrypt/crypto/pkcs11"
+	enchelpers "github.com/containers/ocicrypt/helpers"
+	"gopkg.in/yaml.v2"
+)
+
+// OcicryptConfig represents the format of an imgcrypt.conf config file
+type OcicryptConfig struct {
+	Pkcs11Config pkcs11.Pkcs11Config `yaml:"pkcs11"`
+}
```

Comment:
We now have an environment variables `OCICRYPT_PKCS11_CONF` and a filename `ocicrypt-pkcs11.conf`. So maybe the `OcicryptConfig` should should not exist and we would just use `Pkcs11Config` because at the moment the above requires a config file like this:
```
pkcs11:
  module-directories:
    - /usr/lib64/pkcs11/x
  allowed-module-paths:
    - /usr/lib64/pkcs11/libsofthsm2.so
```

If it's just pkcs11 specific maybe we can just change it to this here:

```
module-directories:
  - /usr/lib64/pkcs11/x
allowed-module-paths:
  - /usr/lib64/pkcs11/libsofthsm2.so
```

Reply:
It may be better to use `OCICRYPT_CONFIG` and only every have one configuration file rather than a pkcs11-specific one and possibly another one... I would rename the filename then to `ocicrypt.conf` as well.